pkg/codegen/schema/schema.go

```diff
@@ -1262,6 +1312,8 @@ func bindConfig(spec ConfigSpec, types *types) ([]*Property, error) {
 }
 
 func bindResource(token string, spec ResourceSpec, types *types) (*Resource, error) {
+	types.resources[token] = &ResourceType{Token: token}
```

Comment:
I think that you'll need to move this into the declaration pass s.t. object types can reference resource types and resources can reference each other.